modules/querying/pages/func/query-user-defined-functions.adoc

```diff
@@ -75,22 +67,39 @@ inline string reverse(string str){
 }
 ----
 
-== Use GitHub to store UDFs
+== Upload UDFs
+
+There are two ways to modify these files to add user-defined functions to GSQL:
+
+* Store UDF files in a GitHub repository, and configure GSQL to read from the repository.
+** This is the recommended approach.
+It also takes the highest precedence.
+If GSQL is configured to read from GitHub for UDFs, and the configurations are valid, local UDF files are ignored.
+* Use `gadmin` commands to upload a local UDF file to GSQL.
+** Use this approach when your server does not have access to the internet and cannot access GitHub.
+
+NOTE: There is a legacy method of xref:func/upload-udf-legacy.adoc[uploading] UDFs through GSQL.
+For security reasons, the legacy method is not recommended.
```

Comment:
Instead of saying it is not recommended because of security reasons, lets say:

_This method is not managed by gadmin and is therefor not recommended_